### PR TITLE
tap-info: include cask tokens and all commands in JSON output

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Follow-up to #7043: 

- also list cask tokens in JSON output (e.g. `brew tap-info homebrew/homebrew-cask --json|jq -r '.[] ' `
- fix logic for detecting command files in taps

### CODE REVIEW

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -422,6 +422,11 @@ def formula_names
     @formula_names ||= formula_files.map { |f| formula_file_to_name(f) }
   end
 
+  # An array of all {Cask} tokens of this {Tap}.
+  def cask_tokens
+    @cask_tokens ||= cask_files.map { |f| formula_file_to_name(f) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]:
```suggestion
    @cask_tokens ||= cask_files.map(&method(:formula_file_to_name))
```

File: Library/Homebrew/tap.rb [hunk 2]
```diff
@@ -472,7 +477,7 @@ def command_dir
   def command_file?(file)
     file = Pathname.new(file) unless file.is_a? Pathname
     file = file.expand_path(path)
-    file.parent == command_dir && file.basename.to_s.match?(/^brew(cask)?-/) &&
+    file.parent == command_dir && file.file? &&
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: Might be nice to reuse some logic from https://github.com/Homebrew/brew/blob/master/Library/Homebrew/commands.rb here?

Comment (EricFromCanada) [reply Library/Homebrew/tap.rb]: With the revamp I figured it was OK to nuke this method entirely.

File: Library/Homebrew/tap.rb [hunk 3]
```diff
@@ -469,17 +475,10 @@ def command_dir
     @command_dir ||= path/"cmd"
   end
 
-  def command_file?(file)
```
Comment (reitermarkus) [inline Library/Homebrew/tap.rb]: This is used by `brew cask ci`.

**Pull request merged**